Fix Exception While Creating an Order in the Admin

justin-at-bounteous opened the pull request:
Changes:
- $total->getDiscountDescription() returns a Phrase, but strlen expects a string
- $total->getDiscountDescription()->getText() will return a string

orlangur commented on app/code/Magento/SalesRule/Model/Quote/Address/Total/ShippingDiscount.php:
```diff
@@ -89,7 +89,7 @@ public function fetch(Quote $quote, Total $total): array
 
         if ($amount != 0) {
-            $description = $total->getDiscountDescription() ?: '';
+            $description = $total->getDiscountDescription() ? $total->getDiscountDescription()->getText() : '';
```
```suggestion
            $description = (string)$total->getDiscountDescription() ?: '';
```

justin-at-bounteous replied:
Changes implemented. Thanks.

orlangur reviewed (changes requested):
After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉

justin-at-bounteous committed:
Fix Exception While Creating an Order in the Admin

Changes:
 - $total->getDiscountDescription() returns a Phrase, but strlen expects a string

magento-engcom-team commented:
Hi @orlangur, thank you for the review.
[ENGCOM-5034](https://jira.corp.magento.com/browse/ENGCOM-5034) has been created to process this Pull Request

stoleksiy commented:
:heavy_check_mark: QA Passed
